pkg/recipes/terraform/module.go

```diff
@@ -73,9 +75,29 @@ func inspectModule(workingDir string, recipe *recipes.EnvironmentDefinition) (*m
 		result.ContextVarExists = true
 	}
 
-	// Extract the list of required providers.
-	for providerName := range mod.RequiredProviders {
-		result.RequiredProviders = append(result.RequiredProviders, providerName)
+	// Extract the details of required providers.
+	for k, v := range mod.RequiredProviders {
+		requiredprovider := &config.RequiredProviderInfo{}
+
+		requiredprovider.Source = v.Source
+		if len(v.VersionConstraints) > 0 {
+			requiredprovider.Version = strings.Join(v.VersionConstraints, ",")
+		}
+
+		if len(v.ConfigurationAliases) > 0 {
+			var aliases []string
+			for _, alias := range v.ConfigurationAliases {
+				// Concatenate Name and Alias from ProviderRef.
+				if alias.Name != "" && alias.Alias != "" {
+					aliases = append(aliases, alias.Name+"."+alias.Alias)
```

Comment:
Could you add a comment with link to the doc referencing why this is needed?

Reply:
updated comment with link.